tuf/api/metadata.py

```diff
@@ -415,53 +415,102 @@ def remove_key(self, role: str, keyid: str) -> None:
+
+        meta: A dictionary that contains information about metadata files::
+            {
+                'meta_file_path_1': <MetaFile INSTANCE_1>.
+                'meta_file_path_2': <MetaFile INSTANCE_2>
+                ...
+            }
+    Attributes:
+        version: An integer indicating the version of the metafile.
+        length: An optional integer indicating the length of the metafile.
+        hashes: An optional dictionary containing hash algorithms and the
+            hashes resulted from applying them over the metafile::
+                'hashes': {
+                    '<HASH ALGO 1>': '<META FILE HASH 1>',
+                    '<HASH ALGO 2>': '<META FILE HASH 2>',
+                    ...
+                }
+
+    """
+
+    def __init__(self, version: int, length: Optional[int] = None,
+            hashes: Optional[JsonDict] = None) -> None:
+        self.version = version
+        self.length = length
+        self.hashes = hashes
+
+
+    def __eq__(self, other: Any) -> bool:
+        """Used to compare objects by their values instead of by their
+        addresses."""
+        if not isinstance(other, MetaFile):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+
+        return self.version == other.version and \
+            self.length == other.length and \
+            self.hashes == other.hashes
+
+
+    def to_dict(self) -> JsonDict:
+        """Returns the JSON-serializable dictionary representation of self. """
+        json_dict = {'version': self.version}
+
+        if self.length is not None:
+            json_dict['length'] = self.length
+
+        if self.hashes is not None:
+            json_dict['hashes'] = self.hashes
+
+        return json_dict
 
 
 class Timestamp(Signed):
     """A container for the signed part of timestamp metadata.
 
     Attributes:
         meta: A dictionary that contains information about snapshot metadata::
-
             {
-                'snapshot.json': {
-                    'version': <SNAPSHOT METADATA VERSION NUMBER>,
-                    'length': <SNAPSHOT METADATA FILE SIZE>, // optional
-                    'hashes': {
-                        '<HASH ALGO 1>': '<SNAPSHOT METADATA FILE HASH 1>',
-                        '<HASH ALGO 2>': '<SNAPSHOT METADATA FILE HASH 2>',
-                        ...
-                    }
-                }
+                'snapshot.json': <MetaFile INSTANCE>
             }
 
     """
     def __init__(
             self, _type: str, version: int, spec_version: str,
             expires: datetime, meta: JsonDict) -> None:
         super().__init__(_type, version, spec_version, expires)
-        # TODO: Add class for meta
-        self.meta = meta
+
+        self.meta = {}
+        self.meta['snapshot.json'] = MetaFile(
+            meta['snapshot.json']['version'],
+            meta['snapshot.json'].get('length'),
+            meta['snapshot.json'].get('hashes'))
```

Comment:
So the way I have designed this in the other classes the constructor does not do any dict-to-object deserialization, but just receives objects and assign them to its attributes. This means the constructor here would stay as it was but with a different type hint.
```python
def __init__(
            self, _type: str, version: int, spec_version: str,
            expires: datetime, meta: Dict[str, Metafile]):
    super().__init__(_type, version, spec_version, expires)
    self.meta = meta
```

And deserialization would happen  in a `from_dict` factory method that looks something like this).
```python
@classmethod
def from_dict(cls, metadata: JsonDict) -> 'Timestamp':
    metadata['meta'] = MetaFile(**metadata['meta']['snapshot.json'])
    return super().from_dict(metadata)
      
```

Please take a look at the constructors and `from_dict` methods of `Metadata` and `Signed` classes for reference (especially the latter has some details in the comments).

Reply:
Let me know if this is unclear. Also, I can see how this is not  immediately obvious when extending the model. I'm happy for suggestions for better development documentation.